geopandas/sindex.py

```diff
@@ -81,22 +223,120 @@ def is_empty(self):
+        # by default, the global set
+        valid_query_predicates = VALID_QUERY_PREDICATES
+
         def __init__(self, geometry):
             # for compatibility with old RTree implementation, store ids/indexes
             original_indexes = geometry.index
             non_empty = geometry[~geometry.values.is_empty]
             self.objects = self.ids = original_indexes[~geometry.values.is_empty]
             super().__init__(non_empty.values.data)
 
+        def query_bulk(self, geometry, predicate=None, sort=False):
+            """Wrapper to expose underlaying pygeos objects to pygeos.query_bulk.
+            This also allows a deterministic (sorted) order for the results.
+            Parameters
+            ----------
+            geometry : {GeoSeries, GeometryArray, numpy.array of PyGEOS geometries}
+                Accepts GeoPandas geometry iterables (GeoSeries, GeometryArray)
+                or a numpy array of PyGEOS geometries.
+            predicate : {None, 'intersects', 'within', 'contains', 'overlaps', 'crosses', 'touches'}, optional
+                If predicate is provided, a prepared version of the input geometry is tested using
+                the predicate function against each item in the index whose extent intersects the
+                envelope of the input geometry: predicate(geometry, tree_geometry).
+            sort : bool, default False
+                If True, results sorted lexicographically using
+                geometry's indexes as the primary key and the sindex's indexes as the
+                secondary key. If False, no additional sorting is applied.
+            Returns
+            -------
+            ndarray with shape (2, n)
+                The first subarray contains input geometry indexes.
+                The second subarray contains tree geometry indexes.
+            See also
+            --------
+            See PyGEOS.strtree documentation for more information.
+            """  # noqa: E501
+
+            if predicate not in self.valid_query_predicates:
+                raise ValueError(
+                    "Got `predicate` = `%s`; " % predicate
+                    + "`predicate` must be one of %s" % self.valid_query_predicates
+                )
+
+            if isinstance(geometry, geoseries.GeoSeries):
+                geometry = geometry.values.data
+            elif isinstance(geometry, GeometryArray):
+                geometry = geometry.data
+            elif not isinstance(geometry, np.ndarray):
+                try:
+                    # accept other iterables, ex list, tuple or generator
+                    iter(geometry)
+                except TypeError:
+                    raise TypeError(
+                        "Got `geometry` of type `%s`, " % type(geometry)
+                        + "`geometry` must be a GeoSeries, GeometryArray, "
+                        "numpy.ndarray or other iterable."
+                    )
```

Comment:
instead of this try/except, I think it is fine to simply convert anything else (which is not a GeoSeries or GeometryArray) to a ndarray with `np.asarray(geometry)`

Reply:
That's a good idea. That should take care of most iterables.

Reply:
Implemented